packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -425,6 +455,32 @@ dependencies:
         XcodeProjectInterpreter: () => xcodeProjectInterpreter,
       });
 
+      // Issue: https://github.com/flutter/flutter/issues/47803
+      testUsingContext('exits the tool if a plugin sets an invalid android package in pubspec.yaml', () async {
+        when(flutterProject.isModule).thenReturn(false);
+        when(androidProject.getEmbeddingVersion()).thenReturn(AndroidEmbeddingVersion.v1);
+
+        createPluginWithInvalidAndroidPackage();
+
+        await expectLater(
+          () async {
+            await injectPlugins(flutterProject);
+          },
+          throwsToolExit(
+            message: 'The plugin plugin1 doesn\'t have a main class defined in '
+                     '/.tmp_rand0/flutter_plugin_invalid_package.rand0/android/src/main/java/plugin1/invalid/UseNewEmbedding.java or '
+                     '/.tmp_rand0/flutter_plugin_invalid_package.rand0/android/src/main/kotlin/plugin1/invalid/UseNewEmbedding.kt. '
```

Comment:
this is probably going to fail on windows

Reply:
It looks like the `MemoryFileSystem` defaults to `/` :)